src/main/java/services/impl/QueryRequesterImpl.java

```diff
@@ -20,10 +23,19 @@
     public String makeQueryRequest(String uri, HttpHeaders headers) {
         RestTemplate restTemplate = new RestTemplate();
         ResponseEntity<String> response =
-                restTemplate.exchange(uri,
-                        HttpMethod.GET,
-                        new HttpEntity<String>(headers),
-                        String.class);
+                null;
+        try {
+            response = restTemplate.exchange(
+                    // Spring framework automatically encodes urls. This ensures we don't pass in an already
+                    // encoded url.
+                    URLDecoder.decode(uri, "UTF-8"),
```

Comment:
definitely the better solution. sorry i did a quick search and didn't find anything fruitful and got lazy. thanks will update